site/content/protocols/mediator-coordination/3.0/readme.md

```diff
@@ -0,0 +1,224 @@
+
+## Connectivity
+
+This protocol consists of three different message requests from the `recipient` that should be replied by the `mediator`:
+
+1. Mediate Request -> Mediate Grant or Mediate Deny
+2. Recipient Update -> Recipient Update Response
+3. Recipient Query -> Recipient
+
+## States
+
+This protocol follows the request-response message exchange pattern, and only requires the simple state of waiting for a response or to produce a response.
+
+## Basic Walkthrough
+
+A `recipient` may discover an agent capable of routing using the Discover Features Protocol 2.0. If protocol is supported with the `mediator`, a `recipient` may send a `mediate-request` to initiate a routing relationship.
+
+First, the `recipient` sends a `mediate-request` message to the `mediator`. If the `mediator` is willing to route messages, it will respond with a `mediate-grant` message, otherwise with a `mediate-deny` message. The `recipient` will share the routing information in the grant message with other contacts.
+
+When a new DID is used by the `recipient`, it must be registered with the `mediator` to enable route identification. This is done with a `recipient-update` message.
+
+The `recipient-update` and `recipient-query` methods are used over time to identify and remove DIDs that are no longer in use by the `recipient`.
+
+## Design By Contract
+
+No protocol specific errors exist. Any errors related to headers or other core features are documented in the appropriate places.
+
+## Security
+
+This protocol expects messages to be encrypted during transmission, and repudiable. 
+
+## Composition
+
+Supported Goal Code | Notes
+--- | ---
+                     |       
+                     |       
+
+
+
+## Message Reference
+
+### Mediate Request
+This message serves as a request from the `recipient` to the `mediator`, asking for the permission (and routing information) to publish the endpoint as a mediator.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/3.0/mediate-request`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/3.0/mediate-request",
+}
+```
+
+### Mediate Deny
+This message serves as notification of the `mediator` denying the `recipient`'s request for mediation.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/3.0/mediate-deny`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/3.0/mediate-deny",
+}
+```
+
+### Mediate Grant
+A mediate grant message is a signal from the `mediator` to the `recipient` that permission is given to distribute the included information as an inbound route.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/3.0/mediate-grant`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/3.0/mediate-grant",
+    "body": 
+            {
+                "routing_did": ["did:peer:z6Mkfriq1MqLBoPWecGoDLjguo1sB9brj6wT3qZ5BxkKpuP6"]
```

Comment:
Should this be a single value? Or should we rename it to routing_dids?

The description indicates it can only be one

Reply:
Nice catch! It looks like the example was changed in the 2.0 version of the protocol after I was making my original edits in commit b3ae932a50c461d1005b335c7c6294f5c862b99c. You do bring up an excellent point, however. In 1.0, the key was called `routing_keys` and was a list. In 2.0, it's called `routing_did` and is a string. I'm personally not aware of any cases where returning multiple dids would be useful, so I'm fine with changing the example to be a single string.